packages/analytics-js/src/components/eventManager/EventManager.ts

```diff
@@ -1,18 +1,263 @@
+   * @param rudderEvent Generated event object
+   * @param options API options
+   * @param callback Callback function
+   * @param pageProps Page properties (available only for page events)
+   */
+  processEvent(
+    rudderEvent: RudderEvent,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    processOptions(rudderEvent as RudderEvent, options);
+    checkForReservedElements(rudderEvent as RudderEvent, this.logger);
+
+    // TODO: Push the generated track event and callback to the event repository
+    // TODO: Handle if device mode integrations are not loaded
+  }
+
+  /**
+   * Generate a 'page' event based on the user-input fields
+   * @param category Page's category
+   * @param name Page name
+   * @param properties Page properties
+   * @param options API options
+   * @param callback Callback function
+   */
+  private processPageEvent(
+    category?: Nullable<string>,
+    name?: Nullable<string>,
+    properties?: Nullable<ApiObject>,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    // TODO: Handle adblocker request
+
+    let props = R.clone(properties) || {};
+    const opts = R.clone(options);
+
+    if (props) {
+      props.name = name;
+      props.category = category;
+    }
+
+    props = getUpdatedPageProperties(props, opts);
+
+    const pageEvent: Partial<RudderEvent> = {
+      properties: props,
+      name,
+      category,
+      type: RudderEventType.PAGE,
+      ...getCommonEventData(props),
+    };
+
+    this.processEvent(pageEvent as RudderEvent, opts, callback);
+  }
+
+  /**
+   * Generate a 'track' event based on the user-input fields
+   * @param event The event name
+   * @param properties Event properties
+   * @param options API options
+   * @param callback Callback function
+   */
+  private processTrackEvent(
+    event: string,
+    properties?: Nullable<ApiObject>,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    const props = R.clone(properties);
+    const opts = R.clone(options);
+    const trackEvent: Partial<RudderEvent> = {
+      properties: props,
+      event,
+      type: RudderEventType.TRACK,
+      ...getCommonEventData(),
+    };
+
+    this.processEvent(trackEvent as RudderEvent, opts, callback);
+  }
+
+  /**
+   * Generate an 'identify' event based on the user-input fields
+   * @param userId New user ID
+   * @param traits User traits
+   * @param options API options
+   * @param callback Callback function
+   */
+  private processIdentifyEvent(
+    userId?: Nullable<string>,
+    traits?: Nullable<ApiObject>,
+    options?: Nullable<ApiOptions>,
+    callback?: Nullable<ApiCallback>,
+  ): void {
+    const cTraits = R.clone(traits);
+    const opts = R.clone(options);
+
+    batch(() => {
+      sessionState.rl_user_id.value = tryStringify(userId);
+      sessionState.rl_trait.value = { ...sessionState.rl_trait.value, ...cTraits };
```

Comment:
Please add TODO here to set the user ID and traits using the `userSession` module to update the storage and session state.

Reply:
That should be handled in the storage module via `effect` on these state parameters.

Reply:
It is already being done here in Analytics class: [here](https://github.com/rudderlabs/rudder-sdk-js/blob/v3-staging/packages/analytics-js/src/components/core/Analytics.ts#L391)

Reply:
Addressed.